src/shared/codelens/codeLensUtils.ts

```diff
@@ -62,9 +62,21 @@ export async function makeCodeLenses({
 
             if (associatedResources.length > 0) {
                 for (const resource of associatedResources) {
+                    let hasApiEvent = false
+                    const events = resource.resourceData.Properties?.Events
+                    if (events) {
+                        // Check for api events
+                        for (const key in events) {
+                            const value = events[key]
+                            if (value.Type === 'Api') {
+                                hasApiEvent = true
```

Comment:
I'm curious, can we do the user one better and let them select which event to create an API config with? In that case, push the `key` into an array and return said array instead of `hasApiEvent`. That way, the user has more control over what they're creating...what's the outputted launch config when selecting an `(API)`  option later?

If you feel like this will add a lot of work to this PR, feel free to consider this another time. Just want to make sure that it's on the list of considerations for the future, especially if users tend to add lots of API events to their templates.

Reply:
Agreed, should be a followup PR. But in this PR, `hasApiEventSource` can meanwhile (in this PR) be changed to `apiEventSources: string[]`, which will contain only 1 item for now.